packages/medusa/src/services/line-item.ts

```diff
@@ -329,6 +295,72 @@ class LineItemService extends TransactionBaseService {
     )
   }
 
+  protected async generateLineItem(
+    variant: {
+      id: string
+      title: string
+      product_id: string
+      product: {
+        title: string
+        thumbnail: string | null
+        discountable: boolean
+        is_giftcard: boolean
+      }
+    },
```

Comment:
Partial variant make all the properties optional whereas here we know the props that are not optional. It also helps to auto-document the API of the method and avoid forgetting to pass some data since Partial will accept anything from a variant and you would have to read the method body to know what is needed as a consumer in case you want to select the required fields from the db and hence improve the db queries. wdyt?

Reply:
Makes sense, my point was more that this is quite verbose but fine to keep as-is :)